backend/src/auth/auth.service.ts

```diff
@@ -74,14 +75,98 @@ export class AuthService {
     };
     const token = await this.jwt.signAsync(payload, {
       expiresIn: '1d',
-      secret: this.config.get('JWT_SECRET')
+      secret: process.env.JWT_SECRET
     });
     return { access_token: token };
   }
   async verifyJwt(token: string) {
     const decoded = await this.jwt.verify(token, {
-      secret: this.config.get('JWT_SECRET')
+      secret: process.env.JWT_SECRET
     });
     return decoded;
   }
+
+  async getFortyTwoProfile(accessToken42: string) {
+    const response = await axios.default
+      .get('https://api.intra.42.fr/v2/me', {
+        headers: {
+          Authorization: `Bearer ${accessToken42}`
+        }
+      })
+      .catch(() => {
+        throw new UnauthorizedException('Invalid 42 token');
+      });
+    return response.data;
+  }
+
+  async fortyTwoAuth(@Res() res: Response, user: any, state: string) {
+    await this.prisma.user
+      .update({
+        where: {
+          id: user.id
+        },
+        data: {
+          api42State: state
+        }
+      })
+      .catch(() => {
+        throw new UnauthorizedException('State is already used');
+      });
+    // TODO: change this to a better way to handle 2fa
+    // backend will never prompt a form, it will be handled by the frontend
+    if (user.twoFactorEnable) {
+      res.send(
+        `<form action="/api/auth/42?state=${state}" method="post"> \
+        <label for="twoFa">Two factor code :</label> \
+        <input id="twoFa" name="twoFa" autofocus></input> \
+        <button type="submit">Submit</button> \
+        </form>`
+      );
+      return;
+    }
```

Comment:
What the F is this?!
Can't we just redirect the user to a 2FA form?

Reply:
You just doesn't like my art :cry:

Reply:
Yes we can ! Thanks